R/upgrade.R

```diff
@@ -39,5 +39,14 @@ dm_upgrade <- function(dm, quiet) {
     dm <- new_dm3(def, zoomed = is_zoomed(dm))
   }
 
+  if (version < 4L) {
+    if (!quiet) {
+      message("Upgrading dm object created with dm <= 1.0.4.") # TODO: Check that package version is correct
+    }
+    def <- unclass(dm)$def
+    def$uuid <- vec_new_uuid_along(def$table)
```

Comment:
We need to adapt that code to make sure we create storage for the autoincrement column here.

Reply:
do we need to add a new version 5? the code could be:
```r
def <- unclass(dm)$def
    def$pks <- map(def$pks, mutate, autoincrement = FALSE) %>%
      vctrs::as_list_of(new_pk())
    dm <- new_dm3(def, zoomed = is_zoomed(dm))
```

Reply:
We only need to update the code that upgrades from version 3 to version 4.

Reply:
done in [0910ce7](https://github.com/cynkra/dm/pull/1654/commits/0910ce7ecb5745ef7d6577c42b951a730459a5d2)